components/twilio/actions/list-messages/list-messages.js

```diff
@@ -0,0 +1,55 @@
+const twilio = require("../../twilio.app.js");
+const { phone } = require("phone");
+
+module.exports = {
+  key: "twilio-list-messages",
+  name: "List Messages",
+  description: "Return a list of messages associated with your account",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    twilio,
+    to: {
+      propDefinition: [
+        twilio,
+        "to",
+      ],
+      description: "Read messages sent to only this phone number. Format the phone number in E.164 format with a + and country code (e.g., +16175551212).",
+      optional: true,
+    },
+    // Not inheriting `propDefinition` from `twilio` because it has value for
+    // `options` property
+    from: {
+      type: "string",
+      label: "From",
+      description: "Read messages sent from only this phone number or alphanumeric sender ID. Format the phone number in E.164 format with a + and country code (e.g., +16175551212).",
+      optional: true,
+    },
+    limit: {
+      propDefinition: [
+        twilio,
+        "limit",
+      ],
+    },
+  },
+  async run() {
+    // Parse the given number into its E.164 equivalent
+    // The E.164 phone number will be included in the first element
+    // of the array, but the array will be empty if parsing fails.
+    // See https://www.npmjs.com/package/phone
+    let to = this.to;
+    if (this.to) {
+      const toParsed = phone(this.to);
+      if (!toParsed || !toParsed.phoneNumber) {
+        throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
+      }
+      to = toParsed.phoneNumber;
+    }
+
+    return this.twilio.listMessages({
+      to: to || undefined, // Use `undefined` if `to` is empty because Twilio API doesn't
```

Comment:
Since you you cant send blank, maybe we can create a helper into `twilio.app.js`to sanitize these values and use here and in all other actions that need to be sanitized?
Something like this:
In `twilio.app.js`

```
...

methods: {
    sanitizeEmptyValues(obj) {
      const data = { ...obj }
      const keys = Object.keys(data);
      for (let i = 0; i < keys.length; i++) {
        if (data[keys] === "") {
          data[keys] = undefined
        }
      }
      return data
    },

...
```

In `list-messages.js` and other actions that need sanitization:
e.g

```
...

return this.twilio.listMessages(this.twilio.sanitizeEmptyValues({
      to,
      from: this.from,
      limit: this.limit,
    }));

...
```
Maybe you can destruct `this.twilio` in order to decrease the line length

Reply:
I created a function [`omitEmptyStringValues`](https://github.com/PipedreamHQ/pipedream/blob/e30dea4129bee325e4eb337aea76677222e7b082/components/twilio/utils.mjs#L32) in `utils.mjs` which returns a new object without entries whose value is `""`.